docs/batch_normalization.rst

```diff
@@ -0,0 +1,397 @@
+>>> vanilla_mlp.initialize()
+>>> x = tensor.matrix('features')
+>>> y = tensor.lmatrix('targets')
+>>> probs = vanilla_mlp.apply(x)
+>>> cost = CategoricalCrossEntropy().apply(y.flatten(), probs)
+>>> cost.name = 'cost'
+>>> error_rate = MisclassificationRate().apply(y.flatten(), probs)
+>>> error_rate.name = 'error_rate'
+
+Try training the network for 10 epochs. The error rate should stay around 90%.
+
+>>> from blocks.algorithms import GradientDescent, Scale
+>>> from blocks.extensions import FinishAfter, Printing
+>>> from blocks.extensions.monitoring import DataStreamMonitoring
+>>> from blocks.graph import ComputationGraph
+>>> from blocks.main_loop import MainLoop
+>>> from fuel.datasets import MNIST
+>>> from fuel.streams import DataStream
+>>> from fuel.schemes import SequentialScheme
+>>> cg = ComputationGraph([cost])
+>>> mnist_train = MNIST('train')
+>>> mnist_test = MNIST('test')
+>>> algorithm = GradientDescent(
+...     cost=cost, params=cg.parameters, step_rule=Scale(0.01))
+>>> main_loop = MainLoop(
+...     algorithm,
+...     DataStream(
+...         mnist_train,
+...         iteration_scheme=SequentialScheme(mnist_train.num_examples, 100)),
+...     extensions=[
+...         FinishAfter(after_n_epochs=10),
+...         DataStreamMonitoring(
+...             [cost, error_rate],
+...             DataStream(
+...                 mnist_train,
+...                 iteration_scheme=SequentialScheme(
+...                     mnist_train.num_examples, 500)),
+...             prefix='train'),
+...         DataStreamMonitoring(
+...             [cost, error_rate],
+...             DataStream(
+...                 mnist_test,
+...                 iteration_scheme=SequentialScheme(
+...                     mnist_test.num_examples, 500)),
+...             prefix='test'),
+...         Printing()])
+>>> main_loop.run() # doctest: +SKIP
+
+Applying batch normalization
+----------------------------
+
+Let's now apply batch normalization to the previous network. We'll normalize
+the output of every linear transformation before non-linearity is applied.
+
+We need to create a slightly different network:
+
+>>> mlp = MLP(
+...     [Sigmoid()] * 4 + [Softmax()], [784] + [500] * 4 + [10],
+...     weights_init=IsotropicGaussian(0.01), use_bias=False)
+>>> mlp.initialize()
+>>> probs = mlp.apply(x)
+>>> cost = CategoricalCrossEntropy().apply(y.flatten(), probs)
+>>> error_rate = MisclassificationRate().apply(y.flatten(), probs)
+
+In this case, we set ``use_bias`` to ``False``, because the batch normalization
+procedure introduces a :math:`\beta` vector that will act as a bias vector.
+
+Let's now retrieve a reference to the symbolic variables that need to be
+normalized:
+
+>>> from blocks.filter import VariableFilter
+>>> from blocks.roles import OUTPUT
+>>> cg = ComputationGraph([cost, error_rate])
+>>> variables = VariableFilter(
+...     bricks=mlp.linear_transformations, roles=[OUTPUT])(cg.variables)
```

Comment:
You are perfectly right...sorry. I looked into `MLP` and my comment doesn't make any sense! I was fooled by the fact that applying the filter with `bricks=[encoding_mlp.linear_transformations]` didn't return any result. It was on a slightly different setting though, I will check if the same issue arises with the code of the example and report it if so.